Library/Formula/qt5.rb

```diff
@@ -72,6 +74,13 @@ class Qt5 < Formula
   depends_on OracleHomeVarRequirement if build.with? "oci"
 
   def install
+    # Save users some frustration by telling them that `--cc=gcc-5` or similar
+    # won't work. It won't be picked up by the Qt build system, there is no easy
+    # way to select an arbitrary compiler, and it's not supported anyway.
+    if ENV.compiler != :clang
```

Comment:
> I guess we don't technically need the repeated `fails_with` unless we know they actually fail.

An actual failure is not the point here. The point is, whatever the user decides to specify via `--cc` is mostly going to be ignored by the Qt build system. As stated in my code comment above I want to save users some frustration if they try to build with a different compiler than Apple Clang and then find out their choice was silently ignored.

I came to this realization and believe it to be true after looking at Qt's `mkspecs` and after some personal experiments. Notice how `??.make.cc` doesn't exist or is very small in a Qt build as it doesn't use the compiler wrappers provided by `superenv`.